satpy/readers/ici_l1b_nc.py

```diff
@@ -274,23 +242,67 @@ def _perform_viewing_angle_interpolation(
                 metadata and the updated dimension names.
 
         """
-        # transform to coordinates where origin is at the equator
-        x, y, z = lonlat2xyz(azimuth, 90. - zenith)
-        lon, lat = xyz2lonlat(x, y, z)
-        lon, lat = self._perform_geo_interpolation(
-            lon,
-            xr.DataArray(lat),
+        # interpolate onto spherical coords system with origin at equator
+        aa, za = self._interpolate_geo(
+            azimuth,
+            90. - zenith,
             n_samples,
         )
-        x, y, z = lonlat2xyz(lon, lat)
-        # transform from cartesian to spherical coords following Eumetsat spec
+        # transform to spherical coords with origin at north pole
+        # following Eumetsat spe
+        x, y, z = lonlat2xyz(aa, za)
```

Comment:
I think this line and the following lines can be removed if you do `za = 90 - za` here, right? or am I misunderstanding this?

Reply:
You are right, this is not needed. Have removed it. Thanks!